### PR TITLE
VIM-5876: Perform Upload Task in Share Extension

### PR DESCRIPTION
**Pull request by supperspidey**
#### Ticket

[VIM-5876](https://vimean.atlassian.net/browse/VIM-5876)

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

This PR assigns the shared container ID to the session configuration before the background session is created.

#### Implementation Summary

N/A

#### Reviewer Tips

N/A

#### How to Test

N/A

### CODE REVIEW

**Comment by supperspidey** [comment]
build this please

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/c9a64a6f1876a7acbf5d05ab14743ee5aeb20587/VimeoUpload/Upload/Descriptor System/Old Upload/OldUploadDescriptor.swift#L50">VimeoUpload/Upload/Descriptor System/Old Upload/OldUploadDescriptor.swift#L50</a></strong>: string interpolation produces a debug description for an optional value; did you mean to make this explicit?  <br /><code>
            print("\(self.currentRequest.rawValue) \(self.identifier)")
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/c9a64a6f1876a7acbf5d05ab14743ee5aeb20587/VimeoUpload/UIKit/DescriptorKVObserver.swift#L31">VimeoUpload/UIKit/DescriptorKVObserver.swift#L31</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorStateDidChange(descriptorType descriptorType: VideoDescriptorType, descriptorState: DescriptorState, isCancelled: Bool, descriptorError: NSError?)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/c9a64a6f1876a7acbf5d05ab14743ee5aeb20587/VimeoUpload/UIKit/DescriptorKVObserver.swift#L32">VimeoUpload/UIKit/DescriptorKVObserver.swift#L32</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorProgressDidChange(descriptorType descriptorType: VideoDescriptorType, progress: Double)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/c9a64a6f1876a7acbf5d05ab14743ee5aeb20587/VimeoUpload/Descriptor System/DescriptorManager.swift#L232">VimeoUpload/Descriptor System/DescriptorManager.swift#L232</a></strong>: conditional downcast from ‘Error?’ to ‘NSError’ is a bridging conversion; did you mean to use ‘as’?  <br /><code>
                let isConnectionError = ((task.error as? NSError)?.isConnectionError() == true || (error as? NSError)?.isConnectionError() == true)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/c9a64a6f1876a7acbf5d05ab14743ee5aeb20587/VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32">VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32</a></strong>: extraneous duplicate parameter name; ‘completion’ already has an argument label  <br /><code>
    func approximateFileSize(completion completion: @escaping FloatBlock)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/c9a64a6f1876a7acbf5d05ab14743ee5aeb20587/VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift#L27">VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift#L27</a></strong>: string interpolation produces a debug description for an optional value; did you mean to make this explicit?  <br /><code>
            print("\(self.currentRequest.rawValue) \(self.identifier)")
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.935 (0.937) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.335 (0.338) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoUpload/263)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: VimeoUpload/Descriptor System/DescriptorManager.swift [hunk 1]
```diff
@@ -317,18 +317,11 @@ open class DescriptorManager: NSObject
     
     // MARK: Public API
     
-    open func handleEventsForBackgroundURLSession(identifier: String, completionHandler: @escaping VoidClosure) -> Bool
+    open func handleEventsForBackgroundURLSession(identifier: String, completionHandler: @escaping VoidClosure)
```
Comment (nicolelehrer) [inline VimeoUpload/Descriptor System/DescriptorManager.swift]: Do we use the passed in identifier anymore? If not, do we need to check that we're handling the right session in the example projects (should we still return here, and if so, should this be called `ifNeeded`?

Comment (supperspidey) [reply VimeoUpload/Descriptor System/DescriptorManager.swift]: @nicolelehrer yes we should remove the `identifier` argument. To help third-party users to check if the descriptor manager should handle background session's events, I've added a new method in `DescriptorManager` for that purpose. I think the library should not absorb the check but instead should let the client decide.

**Comment by supperspidey** [comment]
Thanks @nicolelehrer!

**Pull request merged**